src/Controller/OrderController.php

```diff
@@ -0,0 +1,115 @@
+     * @param Request $request
+     * @return Response
+     */
+    public function submit(Request $request)
+    {
+
+        $message = "order failed";
+        $validator = Validation::createValidator();
+
+        //$content=trim(file_get_contents("php://input"));
+        //$input = json_decode($content, true);
+
+        $input = json_decode($request->getContent(),true);
+
+        $constraint = new Assert\Collection([
+            // the keys correspond to the keys in the input array
+            'price' => new Assert\Type('float'),
+            'year' => new Assert\Type('int'),
+            'make' => [
+                new Assert\Length(['max' => 45]),
+                new Assert\Length(['min' => 10])
+            ],
+            'model' => [
+                new Assert\Length(['max' => 45]),
+                new Assert\Type('string')
+            ],
+            'power' => new Assert\Type('int'),
+            'gearbox-type' => new Assert\Type('string'),
+            'displacement' => new Assert\Type('int'),
+            'ecu' => [
+                new Assert\Length(['max' => 75]),
+                new Assert\Type('string')
+            ],
+            'comment' => new Assert\Type('string'),
+        ]);
+
+        $violations = $validator->validate($input, $constraint);
+
+        if (0 !== count($violations)) {
+            // there are errors, now you can show them
+            foreach ($violations as $violation) {
+                echo $violation->getMessage().'<br>';
+            }
+        }
+        else{
+
+            $em = $this->getDoctrine()->getManager();
+            $order = new Order();
+            $orderVehicle = new OrderVehicleData();
+            $orderState = new OrderState();
+            $user=$this->getDoctrine()->getRepository(User::class)->find(1);
+
+            //$vehicleMake = $this->getDoctrine()->getRepository(VehicleMake::class)->findOneBy(['make' => $input['make']]);
+            $vehicleModel = $this->getDoctrine()->getRepository(VehicleModel::class) ->findOneBy(['model' => $input['model']]);
+
+            $orderVehicle->setYear($input['year']);
+            $orderVehicle->setPower($input['power']);
+            $orderVehicle->setDisplacement($input['year']);
+            $orderVehicle->setEcuModel($input['ecu']);
+            $orderVehicle->setFkVehicleModel($vehicleModel);
+            $em->persist($orderVehicle);
+
+            $em->flush();
+
+            $order->setPriceTotal($input['price']);
+            $order->setComment($input['comment']) ;
+            $order->setFkOrderVehicleData($orderVehicle);
+            $order->setFkUser($user);
+            $em->persist($order);
+
+            $em->flush();
+
+            $orderState->setState('nemodifikuotas');
+            $orderState->setFkOrder($order);
+            $em->persist($orderState);
+
+            $em->flush();
+
+            $message="order successful";
+
+        }
+
+        return $this->render('order/index.html.twig', [
```

Comment:
Endpointas turi grąžinti JSON response:
`return $this->json(['message' => 'yay success!'], Response::HTTP_OK);`